src/agreement/bin_values.rs

```diff
@@ -109,3 +110,17 @@ impl FromIterator<BinValues> for BinValues {
         v
     }
 }
+
+impl IntoIterator for BinValues {
+    type Item = bool;
+    type IntoIter = vec::IntoIter<bool>;
```

Comment:
I propose not to go via `vec` but use statically allocated slices instead:
https://play.rust-lang.org/?gist=17bf2b390e687054a58840d9604acd09&version=stable&mode=debug&edition=2015